app/src/main/java/ar/com/wolox/android/example/ui/login/RootActivity.java

```diff
@@ -0,0 +1,88 @@
+package ar.com.wolox.android.example.ui.login;
+
+import android.content.Context;
+import android.content.Intent;
+import android.content.SharedPreferences;
+
+import androidx.annotation.Nullable;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import ar.com.wolox.android.R;
+import ar.com.wolox.android.example.model.User;
+import ar.com.wolox.android.example.network.UserService;
+import ar.com.wolox.wolmo.core.activity.WolmoActivity;
+import ar.com.wolox.wolmo.networking.retrofit.RetrofitServices;
+import ar.com.wolox.wolmo.networking.retrofit.callback.NetworkCallback;
+import okhttp3.ResponseBody;
+import retrofit2.Call;
+
+/**
+ * RootActivity extends WolmoActivity
+ */
+public class RootActivity extends WolmoActivity {
+
+    @Inject
+    RetrofitServices mRetrofiServices;
+
+    SharedPreferences sharedPref;
+    String email, password, defaultValue;
+    Intent intent;
+
+    @Override
+    protected int layout() {
+        return R.layout.activity_root;
+    }
+
+    @Override
+    protected void init() {
+        sharedPref = this.getSharedPreferences(
+                getString(R.string.preferences_name), Context.MODE_PRIVATE);
+
+        intent = new Intent(this, HomeActivity.class);
+
+        defaultValue = getString(R.string.prefs_default_value);
+        email = sharedPref.getString(getString(R.string.login_email), defaultValue);
+        password = sharedPref.getString(getString(R.string.login_pass), defaultValue);
+
+        if (email.isEmpty() || password.isEmpty()) {
```

Comment:
I don't really like this idea. If there's no sharedPreferences you have to do an intent to LoginActivity not a replaceFragment()